exasol/toolbox/tools/replace_version.py

```diff
@@ -1,34 +1,59 @@
+import re
+from dataclasses import dataclass
+from enum import Enum
 from pathlib import Path
-from typing import List
 
 
-def update_workflow(template: Path, version: str) -> None:
-    """Updates versions of XYZ in GitHub workflow ..."""
+def update_github_yml(template: Path, version: str) -> None:
+    """Updates versions in GitHub workflows and actions"""
     with open(template, encoding="utf-8") as file:
         content = file.readlines()
 
-    content = update_versions(
-        lines=content, matcher="exasol/python-toolbox/.github/", version=version
-    )
+    content = update_versions(lines=content, version=version)
 
     with open(template, "w", encoding="utf-8") as file:
         file.writelines(content)
 
 
-def is_update_required(line, matcher):
-    return matcher in line and "@" in line
+@dataclass(frozen=True)
```

Comment:
A more sophisticated solution would be 
* re-generating the workflows
* enable projects using the PTB and configuring (small) deviations in the workflows (as PK)
* even check for out-dated workflows and display a warning

But this is by far out of scope for the current PR.